packages/openactive-broker-microservice/src/state.js

```diff
@@ -101,78 +101,113 @@ const state = {
   _validatorWorkerPool: null,
   // OPPORTUNITY DATA CACHES
   // We use multiple strategies to cache opportunity data for different use cases.
-  /*
-  TODO, for clarity, consider splitting these caches into categories. I thiiiink
-  that there are three categories: OpportunityIdCache (criteria-oriented),
-  row-ish (rowStoreMap/parentIdIndex), and the rest. It would also be good to
-  more fully understand the latter two categories and see if they are still
-  both needed.
-  */
+  /* TODO investigate consolidation of opportunityItemRowCache and
+  opportunityCache to reduce memory usage & simplify code:
+  https://github.com/openactive/openactive-test-suite/issues/669 */
   /**
    * A criteria-oriented cache for opportunity data. Used to get criteria-matching
    * opportunities for tests.
    */
-  opportunityIdCache: OpportunityIdCache.create(),
-  // nSQL joins appear to be slow, even with indexes. This is an optimisation pending further investigation
+  criteriaOrientedOpportunityIdCache: CriteriaOrientedOpportunityIdCache.create(),
```

Comment:
Maybe a comment linking to the schema of this cache? Just thinking about if someone hovers over `criteriaOrientedOpportunityIdCache` it's not clear what the schema is (which is not the case hovering over other maps)

Reply:
I'll leave this for now because CMD-clicking (in VSCode) will get you there